corehq/apps/data_interfaces/views.py

```diff
@@ -531,10 +521,11 @@ def inaccessible_forms_accessed(self, xform_ids, domain, couch_user):
         return xforms_user_ids - accessible_user_ids
 
     def get_xform_ids(self, request):
-        form_ids_or_query_string = self.get_form_ids_or_query_string(request)
-        if type(form_ids_or_query_string) == list:
-            xform_ids = form_ids_or_query_string
-        elif isinstance(form_ids_or_query_string, basestring):
+        if 'select_all' in self.request.POST:
+            # Altough evaluating form_ids and sending to task would be cleaner,
+            # heavier calls should be in an async task instead
```

Comment:
The bigger fix would be to abstract out the ES query that accepts a querystring and returns form ids, then just call it from here and from the view where it lives now.  That's a bit out of scope, and there may be complications which led the original author of this to choose this request mocking business instead, so I wouldn't worry about it too much.